src/components/Header/Header.js

```diff
@@ -58,7 +58,11 @@ const AddIcon = styled(DiffAdded)`
   }
 `;
 
-const Header = ({ onSetFilter, onStartLogout }) => {
+const Header = () => {
+  const dispatch = useDispatch();
+  const onSetFilter = textValue => dispatch(setFilter(textValue));
+  const onStartLogout = dispatch(startLogout);
```

Comment:
more common solution is to pass function to be dispatched, not the result of dispatch (in similar way as `onSetFilter`. Is there some special reason for this solution?